tests/test_update_password.py

```diff
@@ -0,0 +1,170 @@
+            body=V1Secret(
+                data={"password": b64encode(password)},
+                metadata=V1ObjectMeta(
+                    name=f"user-{name}", labels={LABEL_USER_PASSWORD: "true"}
+                ),
+                type="Opaque",
+            ),
+        ),
+    )
+
+    await coapi.create_namespaced_custom_object(
+        group=API_GROUP,
+        version="v1",
+        plural=RESOURCE_CRATEDB,
+        namespace=namespace.metadata.name,
+        body={
+            "apiVersion": "cloud.crate.io/v1",
+            "kind": "CrateDB",
+            "metadata": {"name": name},
+            "spec": {
+                "cluster": {
+                    "imageRegistry": "crate",
+                    "name": "my-crate-cluster",
+                    "version": "4.1.5",
+                },
+                "nodes": {
+                    "data": [
+                        {
+                            "name": "data",
+                            "replicas": 1,
+                            "resources": {
+                                "cpus": 0.5,
+                                "memory": "1Gi",
+                                "heapRatio": 0.25,
+                                "disk": {
+                                    "storageClass": "default",
+                                    "size": "16GiB",
+                                    "count": 1,
+                                },
+                            },
+                        }
+                    ]
+                },
+                "users": [
+                    {
+                        "name": username,
+                        "password": {
+                            "secretKeyRef": {
+                                "key": "password",
+                                "name": f"user-{name}",
+                            }
+                        },
+                    },
+                ],
+            },
+        },
+    )
+
+    host = await asyncio.wait_for(
+        get_public_host(core, namespace.metadata.name, name),
+        timeout=BACKOFF_TIME * 5,  # It takes a while to retrieve an external IP on AKS.
+    )
+
+    system_password = await get_system_user_password(
+        core, namespace.metadata.name, name
+    )
+    await assert_wait_for(
+        True,
+        is_password_set,
+        host,
+        system_password,
+        SYSTEM_USERNAME,
+        timeout=BACKOFF_TIME * 5,
+    )
+
+    await assert_wait_for(
+        True,
+        is_password_set,
+        host,
+        password,
+        username,
+        timeout=BACKOFF_TIME * 5,
+    )
+
+    await core.patch_namespaced_secret(
+        namespace=namespace.metadata.name,
+        name=f"user-{name}",
+        body=V1Secret(
+            data={"password": b64encode(new_password)},
+        ),
+    )
+
+    await assert_wait_for(
+        True,
+        is_password_set,
+        host,
+        new_password,
+        username,
+        timeout=BACKOFF_TIME * 5,
```

Comment:
This should not take more than a minute or two, IMO. There's shouldn't be any need for retry or whatnot.